File: backend/src/main/java/de/fraunhofer/DigiTales2Go/dataStructure/items/technology/subclasses/TechRequired.java
package de.fraunhofer.DigiTales2Go.dataStructure.items.technology.subclasses;

import de.fraunhofer.DigiTales2Go.dataStructure.dataStructureSubclasses.enumerations.Readiness;
import de.fraunhofer.DigiTales2Go.dataStructure.items.technology.dtos.required.TechRequiredCreationDTO;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

/**
 * The type Tech required.
 *
 * @author Markus Walter
 */
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tech_required")
public class TechRequired {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @Lob
    private String description = "";
    @Lob
    private String useCases = "";
    @Lob
    private String discussion = "";
    @ElementCollection
    @CollectionTable(name = "iao_projects", joinColumns = @JoinColumn(name = "id"))
    @Column(name = "projects")
    @Lob
    private List<String> projectsIAO = new ArrayList<>();
    @Enumerated
    private Readiness readiness = Readiness.NOTSET;


    /**
     * Constrcutor from CreationDTO
     *
     * @param required the required
     */
    public TechRequired(TechRequiredCreationDTO required) {
        this.description = required.getDescription();
        this.discussion = required.getDiscussion();
        this.projectsIAO = required.getProjectsIAO();
        this.readiness = required.getReadiness();
        this.useCases = required.getUseCases();

    }
}

File: frontend/lib/core/detail_objects/technology/tech_required.dart
class TechRequired {
  int id;
  String description;
  String usecases;
  String discussion;
  List<String> projectsIAO;
  String readiness;

  TechRequired({
      required this.id, required this.description, required this.usecases, required this.discussion, required this.projectsIAO, required this.readiness});
  
    factory TechRequired.fromJson(Map<String, dynamic> json) {
    
    return TechRequired(
      id: json['id'] ?? 0,
      description:  json['description'],
      usecases: json['useCases'],
      discussion: json['discussion'],
      projectsIAO: List.from(json['projectsIAO']),
      readiness: json['readiness'],
    );
  }

}
